service/frontend/workflow_handler.go

```diff
@@ -3593,6 +3665,32 @@ func (wh *WorkflowHandler) UpdateWorkerDeploymentVersionMetadata(ctx context.Con
 func (wh *WorkflowHandler) DescribeSchedule(ctx context.Context, request *workflowservice.DescribeScheduleRequest) (_ *workflowservice.DescribeScheduleResponse, retError error) {
 	defer log.CapturePanic(wh.logger, &retError)
 
+	// Prefer CHASM scheduler if enabled.
```

Comment:
Seems like we should first check if CHASM scheduler is enabled to avoid every call to the schedule going through this new code path, especially before this code has been thoroughly tested.

Reply:
We should also move the check if schedules are enabled to the top of this function (or any other function that handles schedules). Users may have disabled the feature completely for some arbitrary reason.